be/src/exec/aggregator.cpp

```diff
@@ -511,6 +511,30 @@ Status Aggregator::_reset_state(RuntimeState* state) {
     return Status::OK();
 }
 
+Status Aggregator::spill_aggregate_data(RuntimeState* state, std::function<StatusOr<ChunkPtr>()> chunk_provider) {
+    auto io_executor = this->spill_channel()->io_executor();
+    auto spiller = this->spiller();
+    auto spill_channel = this->spill_channel();
+
+    while (!spiller->is_full()) {
+        auto chunk_with_st = chunk_provider();
+        if (chunk_with_st.ok()) {
+            if (!chunk_with_st.value()->is_empty()) {
+                RETURN_IF_ERROR(spiller->spill(state, chunk_with_st.value(), *io_executor,
+                                               spill::MemTrackerGuard(tls_mem_tracker)));
+            }
+        } else if (chunk_with_st.status().is_end_of_file()) {
+            return Status::OK();
```

Comment:
when encountering eof , nothing to do? it seems conflusing, suggest to add some comment.